index.bs

```diff
@@ -0,0 +1,331 @@
+Shortname: autoplay-detection
+Level: 1
+Status: w3c/ED
+Group: mediawg
+URL: http://example.com/url-this-spec-will-live-at
+Editor: Alastor Wu, Mozilla https://www.mozilla.org, alwu@mozilla.com
+Editor: Paul Adenot, Mozilla https://www.mozilla.org, padenot@mozilla.com
+Abstract: This specification provides web developers the ability to detect if automatically starting the playback of a media file is allowed in different situations.
+Markup Shorthands: markdown on
+</pre>
+
+<style>
+@media (prefers-color-scheme: light) {
+  :root {
+    --div-info-fg-color: #178217;
+  }
+}
+@media (prefers-color-scheme: dark) {
+  :root {
+    --div-info-fg-color: springgreen;
+  }
+}
+div.enum-description > table {
+	border-collapse: collapse;
+	border-top: 2px solid #707070;
+	border-bottom: 2px solid #707070;
+	width: 100%;
+	margin: 2em 0;
+}
+div.enum-description > table > tbody > tr > th,
+div.enum-description > table > tbody > tr > td {
+	padding: 0.2em 0.6em;
+	min-width: 150px;
+	border-top: 1px solid #ddd
+}
+div.enum-description > table > thead > tr > th {
+	line-height: 2em;
+	font-weight: 600;
+	color: var(--div-info-fg-color);
+	border-bottom: 1px solid #707070;
+}
+</style>
+
+<h2 id="introduction">Introduction</h2>
+  Most user-agents have their own mechanisms to block autoplay media, and those
+  mechanisms are implementation-specific. Web developers need to have a way to
+  detect if the autoplay media is allowed or not in order to make actions, such
+  like selecting alternate content or improving the user experience while media
+  is not allowed to autoplay. For instance, if a user-agent only blocks audible
+  autoplay, then web developers can replace audible media with inaudible media
+  to keep media playing, instead of showing a blocked media which looks like a
+  still image to users. If the user-agent does not allow any autoplay media,
+  then web developers could stop loading media resource and related tasks to
+  save the bandwidth and CPU usage for users.
+
+  Currently, this specification only handles {{HTMLMediaElement}} (<{video}>
+  and <{audio}>) and does not handle [[webaudio inline]], [[speech-api inline]]
+  and <{image}> (GIF animation). Although the result from the document's
+  {{Document/autoplayPolicy}} can be used to detect whether web audio can be
+  started if user-agents block web audio to autoplay.
+
+<h2 id="autoplay-detection-api">The Autoplay Detection API</h2>
+  Autoplay detection can be performed on either the {{Document}} element or on
+  the {{HTMLMediaElement}}. The former provides a general idea to authors to
+  know if autoplay media is allowed on this document element. If the page
+  contains multiple documents, an implementation can decide to return different
+  result for each of them. The latter provides more accurate result for whether
+  the autoplay is allowed on a specific media element.
+
+  <h3 id="autoplay-policy">Autoplay Policy Enum</h3>
+    <pre class="idl">
+    enum AutoplayPolicy {
```

Comment:
Curious if we think this enum may be expanded in the future or if this will be fixed?

Reply:
Nope I don’t at the moment. Just curious if that’s something that would be ok to happen, and it’s related to my other comment on whether media element and web audio will always have values that satisfy both apis.